mozilla/MozDef
# [WIP] Remove doc_type from cron

**Completed:**
- Added logger from mozdef_util and stripped out old logging function with these edits where applicable.
- Update to default template
- Update to rest/index.py for dashboards.
- **healthandstatus** removed doc_type in favor of default elasticsearch_client setting this.
  - Update to defaultMozdefState index mapping
- **healthtomongo**
  - set _type to be "type" = "mozdefhealth"
- **correlateUserMacAddress** sets doc_type to 'usernamemacaddress', change to "_doc"
- **okta2mozdef** removed doc_type in favor of default elasticsearch_client setting this.
- **sqs_queue_status** removed doc_type in favor of default elasticsearch_client setting this.
- **verifyeventfields**
  - removed doc_type in favor of default elasticsearch_client setting this.
  - determine if the check for _type is still needed, or if we want to change it to "type"
    - changed to type for now, can always revert to _type, but with how it excludes mozdefhealth and stats, which are now moved to "type", I felt it was best to preserve the current intended functionality.

**Work To Do:**

- **ESWorkers:**
  - move any custom doc_type being set in any esworker to be a part of the parent structure with the designation of  "type"
  - change actual doc_type in all esworkers workers to be "_doc"
  - **MQ Plugins**
    - Various plugins set doc_type, must be changed to set "type"

- **Alerts:**
  - move any custom doc_types to "type" within alerts.
    - change actual doc_type in all alerts to be "_doc"

- **Cron**
  - **collectattackers**
    - comments say it sets _type, but there was no code that directly sets this that I could find.


- **Elasticsearch Client**
  - **elasticsearch_client.py:** L197: change doc_type to "doc"

- **TESTS**
  - **test_promisc_audit.py**
    - set _type to event or look for "type = auditd" instead
  - **test_write_audit.py**
    - set _type to event or look for "type = auditd" instead
  - **test_cloudtrail_logging_disabled**
    - set _type to event or look for "type = cloudtrail" instead

- **DOCKER**
  - **initial_setup.py**
    - dashboard is no longer "_type", but "type"
    - visualization is no longer "_type", but "type"

- **EXTRA**
  - modify whatever sets empty fields to UNKNOWN to ensure that it no longer sets _type as "UNKNOWN"
  - **Benchmarking**
    - insert_bulk.js needs change _type to "_doc" from "blog"


- **Kibana:**
  - Determine if kibana index name should be cast into a variable
    - This index name appears to get created with a random number (e.g: .kibana_1, kibana_7)
